web/blog/arrow-data-spine/index.md

```diff
@@ -0,0 +1,128 @@
+membership queries). Strong typing at the core of the system allows for
+efficient data representation and enables type-based optimizations, such as
+custom bitmap indexing structures. The data model came first, then the
+representation. Arrow was only a representation:
+
+![MsgPack & Arrow](msgpack-arrow.light.png#gh-light-mode-only)
+![MsgPack & Arrow](msgpack-arrow.dark.png#gh-dark-mode-only)
+
+Early use cases of VAST were limited to interactive, multi-dimensional search to
+extract a subset of disparate records (= rows) in their entirety. The
+row-oriented option worked well for this. But as security operations were
+maturing, requirements extended to analytical processing of structured data,
+making a columnar increasingly beneficial.
+
+Today, the need to bring advanced security analytics and data engineering
+together is stronger than ever, but there is a huge gap between the two fields.
+We see Arrow as the vehicle to close this gap, by practicing *security data
+engineering*. Security analytics should not only be easy and fast. It should
+also allow the expert to act in their domain, end-to-end. This requires with
+treating central domain objects as first-class (e.g., IP addresses, subnets, and
+URLs) and making working with them *easy*. This aspect is where our perception
+of Arrow changed: the primary value proposition of Arrow is to *make data
+interoperability easy*. With an open format specification at its core,
+Arrow-based runtimes were mushrooming all over the place. Nowadays it requires
+only a few lines of code to integrate Arrow data into the central logic of
+applications.
+
+Data interoperability is a sufficient, but not a necessary condition for
+enabling security analytics. We also need to embed the data model of the
+security domain. This is where Arrow's [extension types][extension-types] come
+into play. They add *semantics* to otherwise generic types, e.g., by telling the
+user "this is a transport-layer port" and not just a 16-bit unsigned integer, or
+"this is connection 4-tuple to represent a network flow" instead of "this is
+record with 4 fields of type string and unsigned integer". Extension types are
+composable and allow for creating a rich typing layer with meaningful domain
+objects on top of a standardized data representation. Since they are embedded in
+the data, they do not have to be made available out-of-band when crossing the
+boundaries of different tools. Now we have self-describing security data.
+
+Interoperability plus support for a domain-specific data model makes Arrow a
+solid *data plane*. But Arrow is much more than standardized data
+representation. Arrow also comes with bag of tools for working with the
+standardized data. In the diagram below, we show the various Arrow pieces that
+power the architecture of VAST:
+
+![Arrow Data Plane](arrow-data-plane.light.png#gh-light-mode-only)
+![Arrow Data Plane](arrow-data-plane.dark.png#gh-dark-mode-only)
```

Comment:
Below DataFusion and DuckDB we should mention what we already have: VAST's own query engine.